components/core/src/clp_s/search/OutputHandler.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef CLP_S_SEARCH_OUTPUTHANDLER_HPP
+#define CLP_S_SEARCH_OUTPUTHANDLER_HPP
+
+#include <string>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/exception/exception.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+#include "../Defs.hpp"
+
+namespace clp_s::search {
+class OutputHandler {
+public:
+    explicit OutputHandler(bool output_timestamp, bool decompress_full_log = true)
+            : m_output_timestamp(output_timestamp) {}
+
+    virtual ~OutputHandler() = default;
+
+    virtual void write(std::string const& message, epochtime_t timestamp) = 0;
+    virtual void write(std::string const& message) = 0;
+    virtual void flush() = 0;
+
+    bool output_timestamp() const { return m_output_timestamp; }
+
+protected:
+    bool m_output_timestamp;
+};
+
+class StandardOutputHandler : public OutputHandler {
+public:
+    explicit StandardOutputHandler(bool output_timestamp = false)
+            : OutputHandler(output_timestamp) {}
+
+    void write(std::string const& message, epochtime_t timestamp) override {
+        printf("%lld %s", timestamp, message.c_str());
```

Comment:
We can address the compiler warning here by:
* Defining a new macro in Defs.hpp:
  ```cpp
  #define EPOCHTIME_T_PRINTF_FMT PRId64
  ```
* Replacing this line with:
  ```cpp
  printf("%" EPOCHTIME_T_PRINTF_FMT " %s", timestamp, message.c_str());
  ```